examples/Examples/Common/Example.cs

```diff
@@ -0,0 +1,77 @@
+using CommandLine;
+
+namespace Concordium.Sdk.Examples.Common;
+
+/// <summary>
+/// Represents an example program which supports the command-line
+/// parameters defined by instances of <see cref="ExampleOptions"/>.
+/// </summary>
+public static class Example
+{
+    /// <summary>
+    /// Run an example program specified by a callback and the raw command-
+    /// line parameters. The raw command line parameters are parsed according
+    /// to the supplied <typeparam name="T"/> and the callback is invoked with
+    /// the resulting instance as its argument.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The <see cref="ExampleOptions"/> instance into which <paramref name="args"/>
+    /// will be parsed.
+    /// </typeparam>
+    /// <param name="args">
+    /// The raw command line arguments.
+    /// </param>
+    /// <param name="exampleCallback">
+    /// The callback corresponding to the example program which will be
+    /// invoked with the parsed <typeparam name="T"/> instance as its
+    /// argument.
+    /// </param>
+    public static Task Run<T>(string[] args, Action<T> exampleCallback)
+        where T : ExampleOptions
+    {
+        Task exampleCallbackTask(T options) =>
```

Comment:
The point was not really to start a new `Task`, but since we changed the return types to `Task` meant I had to wrap it like thos. I changed the other method name to `RunAsync`, since this reflects the nature of the method. If we do not have the two, we will have to provide closures for `Action<T>` akin to something like:

```
action.Invoke(options);
await Task.CompletedTask;
```

for each example and/or invoke `Parser` explicitly I guess.

Reply:
Just return the task and remove async